pages/lend/index.tsx

```diff
@@ -378,60 +382,38 @@ const Lend: NextPage = () => {
                 honeyClient,
                 honeyMarket,
                 honeyUser,
-                parsedReserves
+                parsedReserves,
+                mData
               );
 
-              collection.rate =
-                ((await getInterestRate(
-                  collection.utilizationRate,
-                  collection.id
-                )) || 0) * collection.utilizationRate;
-
               collection.stats = getPositionData();
+
               if (currentMarketId == collection.id) {
+                setActiveInterestRate(collection.rate);
                 setActiveMarketSupplied(collection.value);
                 setActiveMarketAvailable(collection.available);
                 setNftPrice(RoundHalfDown(Number(collection.nftPrice)));
+                setFetchedDataObject(collection.marketData[0]);
                 collection.userTotalDeposits
                   ? setUserTotalDeposits(collection.userTotalDeposits)
                   : setUserTotalDeposits(0);
               }
-
-              return collection;
-            } else {
-              await populateMarketData(
-                'LEND',
-                collection,
-                sdkConfig.saberHqConnection,
-                sdkConfig.sdkWallet,
-                currentMarketId,
-                false,
-                [],
-                false
-              );
-
-              collection.rate =
-                ((await getInterestRate(
-                  collection.utilizationRate,
-                  collection.id
-                )) || 0) * collection.utilizationRate;
-
-              collection.stats = getPositionData();
-
-              if (currentMarketId == collection.id) {
-                setActiveMarketSupplied(collection.value);
-                setActiveMarketAvailable(collection.available);
-              }
+              setTimeout(() => {
```

Comment:
ideally we dont use fixed timeouts, but based on loading state

Reply:
true, for now we leave it like this - need to see where the delay is